pkg/orchestration/wiring/datadog/types.go

```diff
@@ -0,0 +1,57 @@
+package datadog
+
+import (
+	"github.com/atlassian/voyager"
+)
+
+type AlarmSpecType string
+type AlarmType string
+
+const (
+	Metric AlarmSpecType = "metric alert"
+)
+const (
+	CPU    AlarmType = "cpu"
+	Memory AlarmType = "memory"
+)
+
+type Alarm struct {
+	Name    string      `json:"name"`
+	Type    string      `json:"type"`
+	Query   string      `json:"query"`
+	Option  AlarmOption `json:"options"`
+	Message string      `json:"message"`
+}
+
+type ServiceInstanceSpec struct {
+	ServiceName voyager.ServiceName `json:"serviceName"`
+	Attributes  Alarm               `json:"attributes"`
+	Environment voyager.EnvType     `json:"environment"`
+	Region      voyager.Region      `json:"region"`
+}
+type AlarmOption struct {
+	Timeout           int             `json:"timeout_h,omitempty"`
```

Comment:
1. Would is be possible to change the field names to camelCase?
2. Don't use `int`, use `int32` or `int64`. See https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#primitive-types
3. `TimeoutSeconds`. See https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#units